solidity/contracts/liquidity-protection/LiquidityProtectionSystemStore.sol

```diff
@@ -0,0 +1,202 @@
+// SPDX-License-Identifier: SEE LICENSE IN LICENSE
+pragma solidity 0.6.12;
+
+import "@openzeppelin/contracts/access/AccessControl.sol";
+import "@openzeppelin/contracts/utils/EnumerableSet.sol";
+import "@openzeppelin/contracts/math/SafeMath.sol";
+
+import "./interfaces/ILiquidityProtectionSystemStore.sol";
+import "../utility/Utils.sol";
+
+/**
+ * @dev This contract aggregates the system balances of the liquidity protection mechanism.
+ */
+contract LiquidityProtectionSystemStore is ILiquidityProtectionSystemStore, AccessControl, Utils {
+    using SafeMath for uint256;
+
+    bytes32 public constant ROLE_SUPERVISOR = keccak256("ROLE_SUPERVISOR");
+    bytes32 public constant ROLE_SEEDER = keccak256("ROLE_SEEDER");
```

Comment:
we don't really need a seeder role here, we can do the same thing we do for the wallet, just a function that migrates everything in one shot.
one function for the limits, another for the minted amounts.

Reply:
Fixed.